### PR TITLE
Part of #17670 Replace Typography with Text component.

### PR DESCRIPTION
**Pull request by PrgrmrHarshShukla**
## Explanation

This PR is a part of the issue #17670 Replace Typography with Text component.

In this PR I have made changes to these files.
- ui\components\app\beta-header\index.js
- ui\components\app\advanced-gas-fee-popover\advanced-gas-fee-defaults\advanced-gas-fee-defaults.js
- ui\components\app\advanced-gas-fee-popover\advanced-gas-fee-gas-limit\advanced-gas-fee-gas-limit.js

### Before
Screenshot (35)

### After
Screenshot (34)

## Pre-merge author checklist

- [ ] I've clearly explained:
  - [ ] What problem this PR is solving
  - [ ] How this problem was solved
  - [ ] How reviewers can test my changes
- [ ] Sufficient automated test coverage has been added

## Pre-merge reviewer checklist

- [ ] Manual testing (e.g. pull and build branch, run in browser, test code being changed)
- [x] PR is linked to the appropriate GitHub issue
- [ ] **IF** this PR fixes a bug in the release milestone, add this PR to the release milestone

If further QA is required (e.g. new feature, complex testing steps, large refactor), add the `Extension QA Board` label.

In this case, a QA Engineer approval will be be required.

### CODE REVIEW

**Comment by brad-decker** [comment]
Hey there! @PrgrmrHarshShukla could you work on the lint issues? 
image

I think this will help get us to some passing tests :)

**Comment by PrgrmrHarshShukla** [comment]
Hello @brad-decker 
Will definitely work on them.
Report to you soon.

**Review by georgewrmarshall — COMMENTED** [review]
Hey @PrgrmrHarshShukla, this is a great start and the code is looking good! Would you be able to add some before/after screenshots of the components themselves? As `AdvancedGasFeeDefaults` and `AdvancedGasFeeGasLimit` don't have stories you will have to get the extension up and running on your local and hunt them down. `BetaHeader` should be a lot easier because you will be able to run storybook `yarn storybook` and find it once Storybook is running on your local by searching for `BetaHeader` in the side panel. Here it is on the develop branch https://metamask.github.io/metamask-storybook/?path=/story/components-app-betaheader--default-story

**Comment by PrgrmrHarshShukla** [comment]
Hello @brad-decker ,

Kindly review #18633 
I have addressed the lint issues in 2 files.
Working on the rest.

**Comment by PrgrmrHarshShukla** [comment]
Hello @georgewrmarshall ,
Thank you.

Definitely going to do that too.

**Comment by PrgrmrHarshShukla** [comment]
Hello @brad-decker 

I worked on the lint issues of the remaining files #18636
Please review it.

**Comment by PrgrmrHarshShukla** [comment]
I made the changes in different PRs.
That might be a problem.

So, I made all the changes to this PR only in the files

- ui\components\app\beta-header\index.js
- ui\components\app\advanced-gas-fee-popover\advanced-gas-fee-defaults\advanced-gas-fee-defaults.js
- ui\components\app\advanced-gas-fee-popover\advanced-gas-fee-gas-limit\advanced-gas-fee-gas-limit.js

Kindly review it now.

**Comment by PrgrmrHarshShukla** [comment]
I have worked on some lint issues.

@brad-decker 
@georgewrmarshall 
Kindly review the changes.

File: ui/components/app/advanced-gas-fee-popover/advanced-gas-fee-gas-limit/advanced-gas-fee-gas-limit.js [hunk 1]
```diff
@@ -58,9 +58,10 @@ const AdvancedGasFeeGasLimit = () => {
   }
 
   return (
-    <Typography
-      tag={TypographyVariant.paragraph}
-      variant={TypographyVariant.H7}
+    <Text
+      tag={TextVariant.bodyMd}
```
Comment (georgewrmarshall) [inline ui/components/app/advanced-gas-fee-popover/advanced-gas-fee-gas-limit/advanced-gas-fee-gas-limit.js:62]: I believe tag for the Typography component is the `as` prop for the Text component so we can remove this
```suggestion
```

File: ui/components/app/advanced-gas-fee-popover/advanced-gas-fee-gas-limit/advanced-gas-fee-gas-limit.js [hunk 2]
```diff
@@ -58,9 +58,10 @@ const AdvancedGasFeeGasLimit = () => {
...
+      variant={TextVariant.bodySm}
+      as="h6"
```
Comment (georgewrmarshall) [inline ui/components/app/advanced-gas-fee-popover/advanced-gas-fee-gas-limit/advanced-gas-fee-gas-limit.js:64]: This is the equivalent of `tag={TypographyVariant.paragraph}` for the Typography component
```suggestion
      as="p"
```

File: ui/components/app/beta-header/index.js [hunk 3]
```diff
@@ -32,8 +32,9 @@ const BetaHeader = () => {
       className="beta-header"
       alignItems={AlignItems.center}
     >
-      <Typography
-        variant={TypographyVariant.H7}
+      <Text
+        variant={TextVariant.bodySm}
+        as="h6"
```
Comment (georgewrmarshall) [inline ui/components/app/beta-header/index.js:37]: We may be able to remove this if it doesn't break any e2e tests. Another contributor made a point that `h6` isn't semantically correct for all text tags
```suggestion
```

**Review by georgewrmarshall — COMMENTED** [review]
Looking good! Made a couple more suggestions

**Comment by georgewrmarshall** [comment]
For the conflict resolution you'll want to choose `import { ButtonIcon, ButtonIconSize, IconName } from '../../component-library';`

**Comment by PrgrmrHarshShukla** [comment]
> For the conflict resolution you'll want to choose `import { ButtonIcon, ButtonIconSize, IconName } from '../../component-library';`

Sure.
Will do.

**Review by georgewrmarshall — COMMENTED** [review]
Hey @PrgrmrHarshShukla, I've fixed the lint issues in this PR. I think it would help if you can get the extension running on your local and the ability to debug lint issues. Also it would be great if you could add screenshots like the below

## Before
Screenshot 2023-04-24 at 8 29 03 AM

### After
Screenshot 2023-04-24 at 8 28 36 AM

These 2 screenshots are taken from running storybook locally `yarn storybook` and taking a screenshot of storybook that is [hosted here](https://metamask.github.io/metamask-storybook/?path=/story/components-app-betaheader--default-story) and built from `develop` branch

**Review by pedronfigueiredo — DISMISSED** [review]

File: ui/components/app/beta-header/index.js [hunk 4]
```diff
@@ -28,8 +32,8 @@ const BetaHeader = () => {
       className="beta-header"
       alignItems={AlignItems.center}
     >
-      <Typography
-        variant={TypographyVariant.H7}
+      <Text
+        variant={TextVariant.bodySm}
```
Comment (DDDDDanica) [inline ui/components/app/beta-header/index.js:36]:
```suggestion
        variant={TextVariant.bodySm}
        as="h6"
```

**Comment by georgewrmarshall** [comment]
Need to update snapshots and also notice that you have made the updates to the Beta header in another file? You can update snapshots using

```
yarn jest --updateSnapshot
```

**Review by georgewrmarshall — APPROVED** [review]

**Comment by georgewrmarshall** [comment]
Hey @PrgrmrHarshShukla, It looks like https://github.com/MetaMask/metamask-extension/pull/18964 was merged before this and made the changes necessary to complete this task. As we don't assign specific issues for https://github.com/MetaMask/metamask-extension/issues/17670 because it is so large I like to apologize. Thanks again for your contribution and I encourage you to continue contributing!

**Pull request closed without merging**